packages/block-library/src/button/save.js

```diff
@@ -6,54 +6,16 @@ import classnames from 'classnames';
 /**
```

Comment:
The way the inheritance works may cause some unexpected behaviors:
- With these changes, if I apply a custom color as a background of a group, all button blocks nested inside don't look like a button (their color is the same as the group). With FSE, I guess there will be many cases where color is applied to a block that contains the posts; in that case, all button blocks created will be "invisible." Although, the "background color" of the button, is applied as background, is it a true "background color" and should inherit the background colors of the parent blocks? Maybe what the button contains is a "main color" or a "primary color" (a different CSS variable) that is applied as background (so the button would not inherit background colors).
- We are setting gradients as --wp--color--gradient so we will have the inheritance of gradients by default. Inheritance of gradients probably looks strange, e.g., a new gradient will be rendered with new start and stops; if the parent has the same gradient, users will notice some discontinuity. I wonder if the inheritance of gradin should be optional? E.g., it makes sense for the button block to inherit a gradient from buttons block, but it does not make sense to inherit from a group. In cases where we don't have an inheritance, we would not use a --wp--color--gradient variable and would set an inline style with the gradient directly. Not using a variable would also not be straightforward. We would need to remove the background color variable when a gradient is set; otherwise, blocks descendants of a block with gradient would not inherit the gradient, but they may end up inheriting the background color from another block above making the gradient useless.